listen_test.go

```
@@ -37,193 +38,111 @@
+	app.hooks.OnPostShutdown(func(err error) error {
+		mu.Lock()
+		defer mu.Unlock()
+		shutdown = true
+		return nil
+	})
+
+	ctx, cancel := context.WithTimeout(context.Background(), 2*time.Second)
+	defer cancel()
 
+	go func() {
 		errs <- app.Listener(ln, ListenConfig{
```

golangci-lint (1.62.2)

Function `Listener` should pass the context parameter

(contextcheck)